cmd/sovereignnode/dataCodec/dataCodec.go

```diff
@@ -0,0 +1,491 @@
+package dataCodec
```

Comment:
This is too large and does too much.
You should have 3 other interfaces (taking `DataCodecProcessor` and splitting it) , each with a specialized behavior in handling data:

- eventData marshaller/serializer
``` go 
	SerializeEventData(eventData sovereign.EventData) ([]byte, error)
	DeserializeEventData(data []byte) (*sovereign.EventData, error)
```

- esdtToken data marshaller/serializer
``` go 
	SerializeTokenData(tokenData sovereign.EsdtTokenData) ([]byte, error)
	DeserializeTokenData(data []byte) (*sovereign.EsdtTokenData, error)
```

- operation data marshaller/serializer
``` go 
	SerializeOperation(operation sovereign.Operation) ([]byte, error)
```

Reply:
All of the above can be contained inside a `SovereignDataDecoder`

Reply:
Created a task for this, can stay as it is for now: https://multiversxlabs.atlassian.net/browse/MX-15286